File: backend/semgrep_rules/tests/javascript/kisa_sw_21_weak_crypto.js
const crypto = require("crypto");
function unsafe(data) {
  // ruleid: kisa.sw21.javascript.weak-cryptographic-algorithm
  return crypto.createHash("md5").update(data).digest("hex");
}
function safe(data) {
  // ok: kisa.sw21.javascript.weak-cryptographic-algorithm
  return crypto.createHash("sha256").update(data).digest("hex");
}

File: backend/semgrep_rules/tests/python/kisa_sw_21_weak_crypto.py
import hashlib
def unsafe(data):
    # ruleid: kisa.sw21.python.weak-cryptographic-algorithm
    return hashlib.md5(data).hexdigest()
def safe(data):
    # ok: kisa.sw21.python.weak-cryptographic-algorithm
    return hashlib.sha256(data).hexdigest()
